[Cairo] Unify all drawing area BackingStore implementation files since the only implementation now is cairo

carlosgcampos opened the pull request:
#### 2599ba15f42a5eb2173858c157caf3972d182bc6
<pre>
[Cairo] Unify all drawing area BackingStore implementation files since the only implementation now is cairo
<a href="https://bugs.webkit.org/show_bug.cgi?id=265375">https://bugs.webkit.org/show_bug.cgi?id=265375</a>

Reviewed by Fujii Hironori.

* Source/WebCore/platform/Cairo.cmake:
* Source/WebCore/platform/SourcesCairo.txt:
* Source/WebCore/platform/graphics/cairo/BackingStoreBackendCairo.h: Removed.
* Source/WebCore/platform/graphics/cairo/BackingStoreBackendCairoImpl.cpp: Removed.
* Source/WebCore/platform/graphics/cairo/BackingStoreBackendCairoImpl.h: Removed.
* Source/WebCore/platform/graphics/cairo/CairoUtilities.h:
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformPlayStation.cmake:
* Source/WebKit/PlatformWin.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.cpp:
(WebKit::DrawingAreaProxyCoordinatedGraphics::paint):
(WebKit::DrawingAreaProxyCoordinatedGraphics::incorporateUpdate):
* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.h:
* Source/WebKit/UIProcess/cairo/BackingStore.cpp:
(WebKit::createCairoImageSurfaceWithFastMalloc):
(WebKit::BackingStore::BackingStore):
(WebKit::BackingStore::paint):
(WebKit::BackingStore::incorporateUpdate):
(WebKit::BackingStore::scroll):
* Source/WebKit/UIProcess/cairo/BackingStore.h:
* Source/WebKit/UIProcess/cairo/BackingStoreCairo.cpp: Removed.
* Source/WebKit/UIProcess/wc/DrawingAreaProxyWC.cpp:
(WebKit::DrawingAreaProxyWC::paint):
(WebKit::DrawingAreaProxyWC::incorporateUpdate):
* Source/WebKit/UIProcess/wc/DrawingAreaProxyWC.h:

Canonical link: <a href="https://commits.webkit.org/271193@main">https://commits.webkit.org/271193@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f87b60dedd207d85b74864026c086260d597ee4c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/27666 "3 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/6304 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/28912 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/29886 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/25289 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/28140 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/8243 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/3697 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/25051 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/27932 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/5060 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/23732 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/4393 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/4560 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/24739 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/30526 "Built successfully") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/24016 "Build is in progress. Recent messages:") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/25261 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/25167 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/30680 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/4582 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/2725 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/28644 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/6079 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/5035 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3573 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/5010 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

fujii commented on Source/WebKit/UIProcess/BackingStore.cpp:
BackingStore.{cpp,h} are only for cairo now. Do we want to move them into UIProcess/cairo directory?

carlosgcampos replied:
Good idea!

fujii commented on Source/WebKit/UIProcess/BackingStore.h:
```diff
@@ -25,51 +25,47 @@
-#include <WebCore/BackingStoreBackendCairo.h>
-#endif
+typedef struct _cairo cairo_t;
+typedef struct _cairo_surface cairo_surface_t;
```
I think we can remove these lines because they are defined in RefPtrCairo.h.

fujii reviewed (approved):
LGTM if EWS is happy.

carlosgcampos committed:
[Cairo] Unify all drawing area BackingStore implementation files since the only implementation now is cairo
https://bugs.webkit.org/show_bug.cgi?id=265375

Reviewed by Fujii Hironori.

* Source/WebCore/platform/Cairo.cmake:
* Source/WebCore/platform/SourcesCairo.txt:
* Source/WebCore/platform/graphics/cairo/BackingStoreBackendCairo.h: Removed.
* Source/WebCore/platform/graphics/cairo/BackingStoreBackendCairoImpl.cpp: Removed.
* Source/WebCore/platform/graphics/cairo/BackingStoreBackendCairoImpl.h: Removed.
* Source/WebCore/platform/graphics/cairo/CairoUtilities.h:
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformPlayStation.cmake:
* Source/WebKit/PlatformWin.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.cpp:
(WebKit::DrawingAreaProxyCoordinatedGraphics::paint):
(WebKit::DrawingAreaProxyCoordinatedGraphics::incorporateUpdate):
* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.h:
* Source/WebKit/UIProcess/cairo/BackingStore.cpp:
(WebKit::createCairoImageSurfaceWithFastMalloc):
(WebKit::BackingStore::BackingStore):
(WebKit::BackingStore::paint):
(WebKit::BackingStore::incorporateUpdate):
(WebKit::BackingStore::scroll):
* Source/WebKit/UIProcess/cairo/BackingStore.h:
* Source/WebKit/UIProcess/cairo/BackingStoreCairo.cpp: Removed.
* Source/WebKit/UIProcess/wc/DrawingAreaProxyWC.cpp:
(WebKit::DrawingAreaProxyWC::paint):
(WebKit::DrawingAreaProxyWC::incorporateUpdate):
* Source/WebKit/UIProcess/wc/DrawingAreaProxyWC.h:

Canonical link: https://commits.webkit.org/271193@main
